src/parse/decl.rs

```diff
@@ -271,11 +271,16 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
             decl.storage_class = StorageClass::Auto;
         }
         if let Some(existing) = self.scope.get_immediate(&decl.id) {
+            let extern_redecl_of_static = match (existing.storage_class, decl.storage_class) {
```

Comment:
Do you need a match here? I think it would be simpler to use `==`.

Reply:
it's a pretty long comparison that rustfmt broke up onto a new line, match keeps it more readable imo